dhar135/picture-perfect
# 15 profile creation

This pull request mainly implemented the profile page setup. It also includes multiple changes to enhance the iOS project configuration, improve the app's logging mechanism, and refactor the routing and theme setup in the Flutter application. The most important changes are summarized below:

# Profile Page Implementation with Image Upload

## Changes Made
- Created EditProfileSheet widget for editing user profile information
- Implemented profile picture upload using Firebase Storage
- Added profile image selection from gallery and camera
- Implemented user profile data update functionality
- Added loading states and error handling

## Key Features
- Profile picture upload with gallery/camera selection
- Real-time profile updates
- User data persistence in Firestore
- Error handling and user feedback
- Loading states during operations

## Technical Details
- Uses `image_picker` for image selection
- Firebase Storage for image storage
- Firestore for user data
- State management with Provider
- Navigation using GoRouter
- Proper state handling with mounted checks

## Implementation Notes
- Profile pictures stored in Firebase Storage under 'profile_pictures/{userId}'
- Image compression and resizing for optimal storage
- Proper error handling for failed uploads
- Safe context usage across async operations

## Security
- Firebase Storage rules implemented for authenticated users
- Secure file access patterns
- User-specific storage paths

## Testing Instructions
1. Navigate to profile page
2. Tap profile picture to edit
3. Select image from gallery or camera
4. Verify upload success and UI update
5. Test error scenarios (no network, invalid file)

## Dependencies Added
```yaml
image_picker: ^1.1.2
```




### iOS Project Configuration:
* Added `IPHONEOS_DEPLOYMENT_TARGET` set to 13.0 in multiple sections of the `project.pbxproj` file to ensure compatibility with iOS 13 and above. [[1]](diffhunk://#diff-042f356aa018014abcea18384290f041306d061ed7f947b1927ce7f653e7f589R480) [[2]](diffhunk://#diff-042f356aa018014abcea18384290f041306d061ed7f947b1927ce7f653e7f589R665) [[3]](diffhunk://#diff-042f356aa018014abcea18384290f041306d061ed7f947b1927ce7f653e7f589R688)
* Included usage descriptions for photo library and camera access in `Info.plist` to comply with iOS privacy requirements.

### Logging Mechanism:
* Introduced a new `AppLogger` class in `lib/src/core/utils/logger.dart` to handle logging throughout the application.

### Routing and State Management:
* Refactored the routing setup by creating a new `AppRouter` class in `lib/src/core/router/app_router.dart` using `go_router` for better route management and transitions.
* Updated `main.dart` to use the new `AppRouter` and added lifecycle management to handle user sign-out when the app is closed.

### Theme Setup:
* Added a new light theme configuration in `lib/src/core/theme/app_theme.dart` for a more flexible and modern UI design.

### Miscellaneous:
* Removed outdated route comments from `main.dart`.

These changes collectively improve the project's structure, enhance user experience, and ensure compliance with platform requirements.
